some documentation and style fixes to smooth_l1_loss

bdhirsh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45588 remove beta defaulting in smooth_l1_loss_backward. added to the bc whitelist
* **#45587 some documentation and style fixes to smooth_l1_loss**

Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

bdhirsh committed:
some documentation and style fixes to smooth_l1_loss

[ghstack-poisoned]

mrshenli commented on aten/src/ATen/native/Loss.cpp:
```diff
@@ -296,17 +296,21 @@ Tensor soft_margin_loss(
 
 Tensor smooth_l1_loss(const Tensor& input, const Tensor& target, const int64_t reduction, double beta) {
-  if (beta <= 0)
+  TORCH_CHECK(beta >= 0, "smooth_l1_loss does not support negative values for beta.")
```
Do we have tests to cover these errors?

bdhirsh replied:
just added one

bdhirsh committed:
Update on "some documentation and style fixes to smooth_l1_loss"


Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

[ghstack-poisoned]

bdhirsh committed:
Update on "some documentation and style fixes to smooth_l1_loss"


Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

[ghstack-poisoned]

bdhirsh committed:
Update on "some documentation and style fixes to smooth_l1_loss"


Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

[ghstack-poisoned]

bdhirsh committed:
Update on "some documentation and style fixes to smooth_l1_loss"


Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

[ghstack-poisoned]

mrshenli commented on test/test_nn.py:
```diff
@@ -7032,6 +7032,10 @@ def test_l1_loss_correct(self):
                 input.abs().mean())
 
+    def test_smoothl1loss_negative_beta_not_supported(self):
+        with self.assertRaises(RuntimeError):
```
nit: you can use `assertRaisesRegex` to also verify that the error message is as expected.

mrshenli reviewed (approved):
LGTM!

gchanan commented on torch/nn/modules/loss.py:
```diff
@@ -780,6 +780,9 @@ class SmoothL1Loss(_Loss):
 
     beta is an optional parameter that defaults to 1.
 
+    Note: When beta is set to 0, this is equivalent to we call out directly to :class:`L1Loss`.
```
this doesn't look grammatically correct, did you mix up two messages?  I would expect something like:

"Note: When beta is set to 0, this is equivalent to calling L1Loss."

Also, is that the correct note doc format?

bdhirsh replied:
thanks, fixing :/ 

I skimmed a few other files containing docs to find the `:class:` anotation- but is there a command that I can run locally to test the formatting?

bdhirsh committed:
Update on "some documentation and style fixes to smooth_l1_loss"


Differential Revision: [D24024313](https://our.internmc.facebook.com/intern/diff/D24024313)

[ghstack-poisoned]
